plugins/zenoh-plugin-storage-manager/src/replication/classification.rs

```diff
@@ -397,8 +401,25 @@ impl SubInterval {
         EventRemoval::NotFound
     }
 
-    fn lookup(&self, event_to_lookup: &EventMetadata) -> Option<&Event> {
-        self.events.get(&event_to_lookup.log_key())
+    /// Looks up the key expression of the provided `event_to_lookup` in this [SubInterval] and,
+    /// depending on its timestamp and an [Event] has been found, returns the [EventLookup].
+    ///
+    /// If the Event in the Replication Log has the same or a greater timestamp than
+    /// `event_to_lookup` then `NewerOrIdentical` is returned. If its timestamp is lower then
+    /// `Older` is returned.
+    ///
+    /// If this SubInterval contains no Event with the same key expression, `NotFound` is returned.
+    pub(crate) fn lookup(&self, event_to_lookup: &EventMetadata) -> EventLookup {
+        match self.events.get(&event_to_lookup.log_key()) {
+            Some(event) => {
+                if event.timestamp >= event_to_lookup.timestamp {
+                    EventLookup::NewerOrIdentical(event)
+                } else {
+                    EventLookup::Older
```

Comment:
Are we sure it will never be useful to get the event in case of an `EventLookup::Older`?
IMO semantically the event exists, we should return it and it should be up to the calling logic to use it or not (unless this has a significant impact on performance).